packages/nextjs-plugin/src/get-hashicorp-packages.ts

```diff
@@ -13,10 +13,25 @@ export function getHashicorpPackages(
 ): string[] {
   let results: string[] = []
 
+  const packageJsonPath = path.join(directory, 'package.json')
+
+  // If we're searching the root directory, check package.json for explicit dependencies. This helps with the case where we are running in a monorepo with dependency hoisting.
+  if (!isRecursive && fs.existsSync(packageJsonPath)) {
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'))
+
+    if (packageJson.dependencies) {
```

Comment:
We aren't checking for `devDependencies` here as in practice they should not be included in application code.